qemu/tests/cpu_info_check.py

```diff
@@ -0,0 +1,146 @@
+import re
+import logging
+
+from avocado.utils import process
+
+from virttest import cpu
+from virttest import env_process
+from virttest import error_context
+from virttest import utils_misc
+
+from virttest.utils_version import VersionInterval
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    cpuinfo query test:
+    1). run query cmd. e.g -cpu ?cpuid
+    2). check the expected info is included in the cmd output.
+    3). Boot guest and check the output of qmp command "qom-list-types"
+    4). Check the output of qmp command "query-cpu-definitions"
+    5). Check the output of qmp command "query-cpu-model-expansion"
+    """
+
+    def remove_models(model_list):
+        """
+        Remove models from cpu_types
+        :param model_list: The list of models to be removed
+        """
+        for model in model_list:
+            try:
+                cpu_types.remove(model)
+            except ValueError:
+                logging.warning('The model to be removed is not'
+                                ' in the list: %s' % model)
+                continue
+
+    def get_patterns(p_list):
+        """
+        Return all possible patterns for given flags
+        :param p_list: The list of flags
+        """
+        r_list = []
+        replace_char = [('_', ''), ('_', '-'), ('.', '-'),
+                        ('.', ''), ('.', '_')]
+        for p in p_list:
+            r_list.extend(list(map(lambda x: p.replace(*x), replace_char)))
+        return set(r_list)
+
+    cpu_types = []
+    list(map(cpu_types.extend, list(cpu.CPU_TYPES.values())))
+
+    qemu_path = utils_misc.get_qemu_binary(params)
+    qemu_version = env_process._get_qemu_version(qemu_path)
+    match = re.search(r'[0-9]+\.[0-9]+\.[0-9]+(\-[0-9]+)?',
+                      qemu_version)
+    host_qemu = match.group(0)
+    shortname = params['shortname']
+    if params.get('os_variant') == 'rhel8':
+        lt_8_2_1 = ('Guest.RHEL.8.2.0' or 'Guest.RHEL.8.1' or
+                    'Guest.RHEL.8.0') in shortname
+        lt_8_3 = lt_8_2_1 or ('Guest.RHEL.8.2' in shortname)
+        if lt_8_3:
+            remove_models(params.objects('cpu_model_8_3_0'))
+        if lt_8_2_1:
+            remove_models(params.objects('cpu_model_8_2_1'))
+        # TODO: Need be updated when there is new cpu model introduced
```

Comment:
Seems we don't need to check the guest `os_variant`, we need to check the host kernel version and host qemu version.
Because we don't need to boot guest if we just check the output of ` -cpu ?`.

Reply:
And current code design will miss the CPU model `EPYC-Rome` if we test guest RHEL.8.2.1.

Reply:
> Seems we don't need to check the guest `os_variant`, we need to check the host kernel version and host qemu version.
> Because we don't need to boot guest if we just check the output of ` -cpu ?`.

Yes you are right, will remove `os_variant` check.

Reply:
> And current code design will miss the CPU model `EPYC-Rome` if we test guest RHEL.8.2.1.

You mean `EPYC-Rome` starts from RHEL.8.2.1? Then cpu_model_8_2_1 should be ' Cooperlake EPYC-Rome', and category `cpu_model_8_3_0` should be removed, right?

Reply:
@nanliu-r Updated, please help check, thanks.